### DISCUSSION TITLE
New Vegas Import/Export

### DISCUSSION THREAD

**Discussion by Stray86** [BadDogSkyrim/PyNifly#178] [Q&A]
Hey there, 

I've been fiddling around trying to nif-bash stuff from New Vegas. My first attempt was just breaking up an outfit down to the chest piece to use as an armor add on. 

I tried pasting the item onto the FNV skeleton extracted from the BSA and the FNV skeleton available here. The skeleton from the BSA results in immediate crash, the skeleton here results in the mesh not rendering in the Geck/game. I'll keep fiddling on my own, but thought someone might have instant understanding of the issue or be able to point me to a guide or resource. 

Update: seems to have just been an options issue. unchecking all import and export options except the appropriate rename and importing the skeleton first instead of using the reference skeleton.

**Comment by DarianLStephens** [comment]
I've had some success in NV imports and exports, but am struggling with actual animations. Nifs are pretty much fine, but I haven't managed to get .kf files to work.
One of the issues is that I don't see how to set a bone priority on export with PyNifly. This is essential for NV animations, as it tells the game what animation takes priority on a given bone, based on the number. I could easily be missing it somewhere, but it doesn't appear that PyNifly allows you to set that, so it's always 0 and you need to edit all the transform controllers individually in Nifskope to give them a proper priority.

**Comment by BadDogSkyrim** [comment]
NV support is really minimal. I'm a little surprised you got as far as you did. I don't know if bone priority is a thing in later games--there's another mechanism of auxbones and "blend" animation that maybe takes its place. 

I would imagine there'd be a bunch of other differences in the NV animation format as well.

**Reply by DarianLStephens** [reply]
There aren't as many as you might expect. kf files are, for the most part, functional, there are just a few things which keep it from working out-of-the-box on export.
Older Niftools addons allowed you to set the priority for bones directly, via a new Nif panel on each bone, but perhaps you don't want to go that far for something that isn't a focus of support?

I would absolutely love to have this support NV better. Even with the limitations it has, it's still very good, and I've gotten a lot working ingame via this tool.
Aside from the animations, one particular pain point is the lack of body partition data on export. It's similar to the segments in Fo4, and is also used for dismemberment. They don't really have names, but being able to use some prefix on vertex groups to have the tool export them as NiSkinPartition blocks, with the strips assigned (The only thing really missing, I think), and the body part as the name, would be a very big help to being able to export creatures and other skinned nifs with dismemberment.
Something like 'SPD BP_RIGHTLEG', so it knows that this group is partition data and what body part to assign it to.

Here is where it's all just the default BP_TORSO:
<img width="821" height="598" alt="image" src="https://github.com/user-attachments/assets/e57c6613-bb2d-4b9d-aa70-9eaea8d1f5ff" />

And here is where you'd set the actual strip data:
<img width="563" height="660" alt="image" src="https://github.com/user-attachments/assets/a75c0eeb-498e-4451-a5d6-fe14726a7633" />

For this, I just imported and exported the NV Gecko, and it came out without any partition data. The vertex groups seem to exist in Blender, but as 'unknown' data, and don't get exported:
<img width="1706" height="828" alt="image" src="https://github.com/user-attachments/assets/04e498d4-bb5c-494b-940a-1ec387a823d1" />

Compare to the vanilla nif fresh out of the archive:
<img width="530" height="647" alt="image" src="https://github.com/user-attachments/assets/5dac4b47-093a-4770-b5f8-c04bf17a023f" />

<img width="578" height="1044" alt="image" src="https://github.com/user-attachments/assets/f2201838-c6b8-425d-bd5b-ff18f849aceb" />

I don't know if you're very interested in this, but having better tools for working with nifs and .kf files in New Vegas would be so very nice, as the only alternatives we have are extremely old Blender versions, because that's when the first working tools were made.
I'd be willing to work with you on this, if this is something you would be okay with pursuing.

**Reply by BadDogSkyrim** [reply]
NV was pretty popular in its day but I've never played or owned it. Maybe the next time it comes up on sale I'll buy it and add more support.

**Reply by DarianLStephens** [reply]
It goes on sale pretty regularly. Just a few weeks back it was on sale for, I think a dollar? For the entire game + DLCs.
Heck, I'd buy it for you at that price. If Steam allowed it, I'd buy a whole bunch of it to keep in my inventory and give out to people.

The game still has a very active modding community, but suffers from a lack of modern tools.

**Reply by DarianLStephens** [reply]
Yeah, I'm _really_ trying to get this to work, and I've now gotten as far as having playable animations exported using this addon, with some Nifskope edits, but...
<img width="303" height="309" alt="image" src="https://github.com/user-attachments/assets/4b6dbb0d-c03a-4606-8798-f2dd0da39ffd" />
<img width="220" height="291" alt="image" src="https://github.com/user-attachments/assets/4a5403ff-e007-4b37-9e1c-78126a3c2100" />
The bones are rotated entirely the wrong way. The animation is visually present, and seems to move as expected, but the orientation of the bones is incorrect.
For reference, this is the creature with no animations:
<img width="591" height="984" alt="image" src="https://github.com/user-attachments/assets/59956a0b-a902-4bd4-ba04-16c622809379" />

I've tried exporting the skeleton and nif with the 'Blender orientation' selected, but didn't see any difference in the actual bone rotations, not was there any option to, say, set an 'up' direction when exporting the .kf file.
I also had to change the 'User Version' in the kf from 12 to 11 before the game would play it, but that seems super easy to fix.

**Comment by BadDogSkyrim** [comment]
Okay so... I bought the game. No guarantees about PyNifly support but at least I have the assets.

Best thing to do at this point is turn off all the convenience bone handling. No create bones, rename bones, blender orientation, pretty orientation, anything else I haven't thought of. Same on export. No need to import the armature separately, just use the bones that came with the nif.

**Reply by DarianLStephens** [reply]
That's what I've been doing. The bones are already part of the nif, and since it's a completely custom creature, I've been turning off all the auto bone features on export, but the animation still came out bad.
We could talk more directly on Discord, unless you're fine continuing here. I can send you the nifs, kfs, and .blend files to examine and explain what's wrong and what's needed.

Thank you for doing this, by the way! It's been a bit sad to see NV get left behind by all these great new tools, the possibility of getting some support is very nice!

**Comment by BadDogSkyrim** [comment]
Right now I'm deep in another project but once that's sorted I can have a look at this.
